### DISCUSSION TITLE
Does it refresh embeddings upon note changes?

### DISCUSSION THREAD

**Discussion by preslavrachev** [brianpetro/obsidian-smart-connections#550] [Q&A]
I have recently started using the Smart Connections plugin, and the initial indexing went fine. However, I don't seem to be seeing any updates when I change the existing notes. I have a couple of notes that serve as running drafts, and I keep updating them every day. I thought that the plugin would note changes and update embeddings accordingly, but it does not seem to be the case. Or, is it?

**Comment by brianpetro** [comment]
Hi @preslavrachev

Thanks for trying Smart Connections.

There is a threshold (~10% file size change) that might be preventing the update. Improving this threshold is something I've been thinking about.

One option is, with the recently changed note open, you can use the "Find connections" command in the command-palette. This _should_ update the embedding for the currently opened note. If that doesn't seem to work, let me know because it might have been overlooked during updates.

Another option is the "force refresh" option in the settings (it might depend on whether you're on the early or general release, but the `v2.1` early release should be pushed to the general release today).

🌴

**Comment by quicly** [comment]
@brianpetro 

> One option is, with the recently changed note open, you can use the "Find connections" command in the command-palette. This should update the embedding for the currently opened note. If that doesn't seem to work, let me know because it might have been overlooked during updates.

For me, it doesn't work. If I select the text inside the note and use the command "find smart connections," it finds similar notes for the excerpt. But if I try to apply this command to the whole note, nothing changes even after the content of the note has been replaced.

**Reply by brianpetro** [reply]
I noticed this, too. It's a bug that will need to be fixed.

Sorry for the confusion and thanks for letting me know
🌴
